src/cmd/singularity/cli/singularity.go

```diff
@@ -56,13 +56,14 @@ func init() {
 	}
 	defaultTokenFile := path.Join(usr.HomeDir, ".singularity", "sylabs-token")
 	// authToken priority default_file < env < file_flag
-	authToken = auth.ReadToken(defaultTokenFile)
 	SingularityCmd.Flags().StringVar(&tokenFile, "tokenfile", defaultTokenFile, "path to the file holding your sylabs authentication token")
 	if val := os.Getenv("SYLABS_TOKEN"); val != "" {
 		authToken = val
 	}
 	if i := strings.Compare(tokenFile, defaultTokenFile); i != 0 {
-		authToken = auth.ReadToken(tokenFile)
+		authToken, authWarning = auth.ReadToken(tokenFile)
+	} else {
+		authToken, authWarning = auth.ReadToken(defaultTokenFile)
```

Comment:
Don't think this will work for the priority intended? The defaultTokenFile is now going to override something set by the env var if I'm reading this correctly.